crazyflieTypes:
  default:
    bigQuad: False
    batteryVoltageWarning: 3.8 # V
    batteryVoltateCritical: 3.7 # V
    markerConfiguration: 1 # change to 3 for single marker tracking
    dynamicsConfiguration: 0
    firmwareParams:
      ctrlMel:
        kp_xy: 0.4 #0.6 # p for position
        kd_xy: 0.2 # 0.35 # d for position
        ki_xy: 0.05
        i_range_xy: 2.0
        kR_xy: 70000 # p for attitude
        kw_xy: 20000 # d for attitude
        kR_z: 60000 # 70000 # p for yaw
        kw_z: 12000 # 30000  # d for yaw
        ki_m_z: 500 #0 #20000     # i for yaw
        i_range_m_z: 1500 # i range for yaw
        kd_omega_rp: 200 # roll and pitch angular velocity d gain
        kp_z: 1.25
        kd_z: 0.4
        ki_z: 0.05
        i_range_z: 0.4
        mass: 0.032
        massThrust: 132000
      posCtlPid:
        xKi: 0.0
        yKi: 0.0
        zKi: 0.0
      pid_attitude:
        roll_ki: 0.0
        pitch_ki: 0.0
      pid_rate:
        roll_ki: 0.0
        pitch_ki: 0.0
      velCtlPid:
        vxKi: 0.0
        vyKi: 0.0
        vzKi: 0.0

      # Default PID Attitude Values
      # pid_attitude:
      #   roll_kp: 5.0
      #   roll_ki: 0.0
      #   roll_kd: 2.0
      #   pitch_kp: 5.0
      #   pitch_ki: 0.10
      #   pitch_kd: 2.0

  # leftMarker:
  #   bigQuad: False
  #   batteryVoltageWarning: 3.8 # V
  #   batteryVoltateCritical: 3.7 # V
  #   markerConfiguration: 6 # change to 3 for single marker tracking
  #   dynamicsConfiguration: 0
  #   firmwareParams:

  rightMarker:
    bigQuad: False
    batteryVoltageWarning: 3.8 # V
    batteryVoltateCritical: 3.7 # V
    markerConfiguration: 7 # change to 3 for single marker tracking
    dynamicsConfiguration: 0
    firmwareParams:
      posCtlPid:
        xKi: 0.0
        yKi: 0.0
        zKi: 0.0
      pid_attitude:
        roll_ki: 0.0
        pitch_ki: 0.0
      pid_rate:
        roll_ki: 0.0
        pitch_ki: 0.0
      velCtlPid:
        vxKi: 0.0
        vyKi: 0.0
        vzKi: 0.0

  CF21SingleMarker:
    bigQuad: False
    batteryVoltageWarning: 3.8 # V
    batteryVoltateCritical: 3.7 # V
    markerConfiguration: 3 # changed to 3 for single marker tracking
    dynamicsConfiguration: 0
    firmwareParams:
      kalman:
        pNAcc_xy: 1.0 # default 0.5
        pNAcc_z: 2.0 # default 1.0
        mNGyro_rollpitch: 0.1 # default 0.1
        mNGyro_yaw: 0.1 # default 0.1
      ctrlMel:
        kp_xy: 0.4 #0.6 # p for position
        kd_xy: 0.2 # 0.35 # d for position
        ki_xy: 0.05
        i_range_xy: 2.0
        kR_xy: 70000 # p for attitude
        kw_xy: 20000 # d for attitude
        kR_z: 60000 # 70000 # p for yaw
        kw_z: 12000 # 30000  # d for yaw
        ki_m_z: 500 #0 #20000     # i for yaw
        i_range_m_z: 1500 # i range for yaw
        kd_omega_rp: 200 # roll and pitch angular velocity d gain
        kp_z: 1.25
        kd_z: 0.4
        ki_z: 0.05
        i_range_z: 0.4
        mass: 0.032
        massThrust: 132000

numMarkerConfigurations: 10
markerConfigurations:
  "0": # for standard Crazyflie
    numPoints: 4
    offset: [0.0, -0.01, -0.04]
    points:
      "0": [0.0177184, 0.0139654, 0.0557585]
      "1": [-0.0262914, 0.0509139, 0.0402475]
      "2": [-0.0328889, -0.02757, 0.0390601]
      "3": [0.0431307, -0.0331216, 0.0388839]
  "1": # lightWeight_V6
    numPoints: 4
    offset: [0.0, 0.0, -0.03]
    points:
      "0": [-0.0504837, 0.0150054, 0.0363401]
      "1": [0.00309327, -0.0392855, 0.0393555]
      "2": [-0.0504698, 0.0748615, 0.0382872]
      "3": [-0.0540941, -0.0387714, 0.0372181]

  "2": # big frame
    numPoints: 4
    offset: [0.0, 0.0, -0.06]
    points:
      "0": [0.0558163, -0.00196302, 0.0945539]
      "1": [-0.0113941, 0.00945842, 0.0984811]
      "2": [-0.0306277, 0.0514879, 0.0520456]
      "3": [0.0535816, -0.0400775, 0.0432799]
  "3": # new lightweight cage
    numPoints: 4
    offset: [0.0, 0.0, -0.02]
    points:
      "0": [0.0638817, 0.0114961, 0.0423504]
      "1": [-0.0549325, 0.0690123, 0.0439131]
      "2": [0.0634328, 0.0678059, 0.0441438]
      "3": [0.00515003, -0.0463719, 0.0393996]
  "4": # front facing touch marker
    numPoints: 4
    offset: [0.0, 0.0, -0.06]
    points:
      "0": [-0.00320672, 0.0711399, 0.0739474]
      "1": [0.0546534, 0.0643949, 0.0739337]
      "2": [0.0385261, -0.0461827, 0.0722514]
      "3": [0.0654391, -0.0196962, 0.0727022]
  "5": # front facing touch marker
    numPoints: 4
    offset: [0.0, 0.0, -0.02]
    points:
      "0": [0.0779831, 0.076849, 0.0526226]
      "1": [-0.0390897, 0.0779268, 0.0514557]
      "2": [0.0190234, 0.0792278, 0.0505197]
      "3": [0.079656, 0.011165, 0.0479861]

  "6": # lightweight cage left facing
    numPoints: 4
    offset: [0.0, 0.0, -0.05]
    points:
      "0": [0.0620779, 0.0558468, 0.125515]
      "1": [0.0820516, 0.0265445, 0.125525]
      "2": [0.0480958, 0.0247824, 0.126187]
      "3": [0.0632508, 0.000276077, 0.126366]

  "7": # right 24/2/1
    numPoints: 4
    offset: [0.0, 0.0, -0.04]
    points:
      "0": [-0.0410349, -0.0502113, 0.0402004]
      "1": [0.0170354, -0.0509094, 0.0418944]
      "2": [-0.0423259, 0.00790672, 0.0407434]
      "3": [0.0740889, -0.0487123, 0.0462185]
  "8": # standard without cage
    numPoints: 4
    offset: [0.0, 0.0, -0.02]
    points:
      "0": [0.007879, 0.0336344, 0.0243017]
      "1": [-0.00232167, 0.0194679, 0.0243211]
      "2": [0.0329582, 0.0143487, 0.0224059]
      "3": [0.0076539, -0.0128706, 0.0250476]
  "9": # assemble light weight
    numPoints: 4
    offset: [0.0, 0.0, -0.02]
    points:
      "0": [0.0638817, 0.0114961, 0.0423504]
      "1": [-0.0549325, 0.0690123, 0.0439131]
      "2": [0.0634328, 0.0678059, 0.0441438]
      "3": [0.00515003, -0.0463719, 0.0393996]

numDynamicsConfigurations: 1
dynamicsConfigurations:
  "0":
    maxXVelocity: 2.0
    maxYVelocity: 2.0
    maxZVelocity: 3.0
    maxPitchRate: 20.0
    maxRollRate: 20.0
    maxYawRate: 10.0
    maxRoll: 1.4
    maxPitch: 1.4
    maxFitnessScore: 0.001
